flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointRequestDeciderTest.java

```diff
@@ -119,13 +119,13 @@ public void testQueueSizeLimitPriority() {
 	public void testSavepointTiming() {
 		testTiming(regularSavepoint(), TriggerExpectation.IMMEDIATELY);
 		testTiming(periodicSavepoint(), TriggerExpectation.IMMEDIATELY);
-		testTiming(nonForcedSavepoint(), TriggerExpectation.AFTER_PAUSE);
+		testTiming(nonForcedSavepoint(), TriggerExpectation.IMMEDIATELY);
```

Comment:
after this change, do we need to add a test to guarantee "savepoint and checkpoint would not execute concurrently under unaligned checkpoint mode"

Reply:
Yes, @klion26 , maybe we need this. 
There are two functions called  `testUserSubmittedPrioritized` and `testSavepointPrioritized` in `CheckpointRequestDeciderTest` for testing the ordering of requests', forced savepoint and checkpoints, and I will write two other test functions for testing the order of unforced savepoints'  and checkpoints' requests , using `testRequestsOrdering` too.
What do you think about this, @klion26

Reply:
I think maxPendingCheckpoints limit (what @klion26 proposed) and order are different things to check.
For the 1st one, I think we can reuse `testEnqueueOnTooManyPending` by calling `nonForcedSavepoint` instead of  `regularCheckpoint`.

Reply:
thanks, I will add a test function reusing `testEnqueueOnTooManyPending` as you say.

Reply:
and I want also add two other test functions for testing order, too,  what do you think about this, will they become redundant?

Reply:
Hi, @klion26  @rkhachatryan , I add three test functions:
* testNonForcedEnqueueOnTooManyPending: testing for nonforced savepoint on too many pending
* testNonForcedUserSubmittedPrioritized: testing for order of nonforced savpoint and nonForcedPeriodicSavepoint
* testNonForcedSavepointPrioritized: testing for order of nonForcedSavepoint and regularCheckpoint